src/Instrumentation/Symfony/Messenger/InstrumentationEventSubscriber.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace FriendsOfOpenTelemetry\OpenTelemetryBundle\Instrumentation\Symfony\Messenger;
+
+use FriendsOfOpenTelemetry\OpenTelemetryBundle\Instrumentation\InstrumentationTypeEnum;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\API\Trace\TracerInterface;
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\SDK\Trace\Span;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Messenger\Event\WorkerMessageFailedEvent;
+use Symfony\Component\Messenger\Event\WorkerMessageHandledEvent;
+use Symfony\Component\Messenger\Event\WorkerMessageReceivedEvent;
+use Symfony\Component\Messenger\Stamp\BusNameStamp;
+
+/**
+ * Be aware if you start a span before this subscriber, it could leads to orphan span issue.
+ * Be sure your span is properly ended.
+ */
+class InstrumentationEventSubscriber implements EventSubscriberInterface
+{
+    private ?InstrumentationTypeEnum $instrumentationType = null;
+
+    public function __construct(
+        private readonly TracerInterface $tracer,
+        private readonly LoggerInterface $logger,
+    ) {
+    }
+
+    public function setInstrumentationType(InstrumentationTypeEnum $instrumentationType): void
+    {
+        $this->instrumentationType = $instrumentationType;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            WorkerMessageReceivedEvent::class => ['startSpan'],
+            WorkerMessageFailedEvent::class => ['endSpanOnError'],
+            WorkerMessageHandledEvent::class => ['endSpanWithSuccess'],
+        ];
+    }
+
+    public function startSpan(WorkerMessageReceivedEvent $event): void
+    {
+        if (InstrumentationTypeEnum::Auto !== $this->instrumentationType) {
```

Comment:
I'd like to adapt this to handle manual instrumentation, for example if the message has the `#[Traceable]` attribute, it should be instrumented.